src/github.com/getlantern/balancer/balancer.go

```diff
@@ -49,7 +68,25 @@ func New(dialers ...*Dialer) *Balancer {
 // remaining Dialers until it either manages to connect, or runs out of dialers
 // in which case it returns an error.
 func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error) {
-	dialers := b.dialers
+	var dialers []*dialer
+
+	// Checking destination port.
+	_, port, _ := net.SplitHostPort(addr)
+
+	// Are we attempting to connect to port 80 (plain HTTP)?
```

Comment:
The `8080` looks like a really good candidate here, but I'm not sure about the other ports, `5222` and `5223` are commonly used for XMPP and `5222` is the plaintext one so maybe we should add it too. And what about `5228`? Why does Lantern support it?

One thing to keep in mind is that we cannot guarantee that Lantern will identify and protect unencrypted traffic automatically: users may as well use plain HTTP over port 443 on specially crafted servers and Lantern will assume that if it's using port 433 it should be SSL, also using port 8443 does not guarantee that the connection will be over SSL, maybe we can start by only proxying ports such as 80 and 8080 on our servers since that would fix the potential Javascript script injection we're trying to close, then we could discuss about supporting plaintext traffic over other ports.